swap_meet/vendor.py

```diff
@@ -1,2 +1,73 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory
+        if self.inventory is None:
+            self.inventory = []
+
+    def add(self, item):
+        '''adds item to inventory'''
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        '''removes item from inventory'''
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        '''returns list of items of a given category from inventory'''
+        same_category_items = []
+        for item in self.inventory:
+            if item.category == category:
+                same_category_items.append(item)
+        return same_category_items
+
+    def swap_items(self, vendor, my_item, their_item):
+        '''swaps items between vendor inventories'''
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        else:
+            self.inventory.remove(my_item)
+            vendor.inventory.append(my_item)
+            vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
```

Comment:
Great logic! One small thing: the `else` is unneeded here because the `if` block returns and ends the function if it's entered.